Dockerfile

```diff
@@ -0,0 +1,49 @@
+
+FROM php:8.0-fpm
+
+# Copy composer.lock and composer.json
+
+# Set working directory
+WORKDIR /var/www
+
+# Install dependencies
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    libzip-dev \
+    libpng-dev \
+    libjpeg62-turbo-dev \
+    libwebp-dev libjpeg62-turbo-dev libpng-dev libxpm-dev \
+    libfreetype6 \
+    libfreetype6-dev \
+    locales \
+    zip \
+    libonig-dev \
+    jpegoptim optipng pngquant gifsicle \
+    vim \
+    unzip \
+    git \
+    curl
+
+# Clear cache
+RUN apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# Install extensions
+RUN docker-php-ext-install pdo_mysql mbstring zip exif pcntl
+RUN pecl install xdebug \
+    && docker-php-ext-enable xdebug
+
+# Install composer
+RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
```

Comment:
Instead of running composer and installing into this container for 1 file, we can do a docker multi-stage build for this to grab the composer file from the pure docker iteration - 

```
COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
```